test/NuGet.Core.Tests/NuGet.Client.Test/ContentModelContentFilesTests.cs

```diff
@@ -0,0 +1,174 @@
+
+            // Assert
+            Assert.Equal(6, contentFileGroups.Count());
+
+            Assert.Equal("any|cs|vb", string.Join("|",
+                contentFileGroups.Select(group =>
+                    group.Properties[ManagedCodeConventions.PropertyNames.CodeLanguage])
+                    .Distinct()
+                    .OrderBy(s => s)));
+
+            Assert.Equal("net45|uap", string.Join("|",
+                contentFileGroups.Select(group =>
+                    group.Properties[ManagedCodeConventions.PropertyNames.TargetFrameworkMoniker] as NuGetFramework)
+                    .Select(f => f.GetShortFolderName())
+                    .Distinct()
+                    .OrderBy(s => s)));
+        }
+
+        [Fact]
+        public void ContentFiles_FrameworkNormalize()
+        {
+            // Arrange
+            var conventions = new ManagedCodeConventions(
+                new RuntimeGraph(
+                    new List<CompatibilityProfile>() { new CompatibilityProfile("net46.app") }));
+
+            var criteria = conventions.Criteria.ForFramework(NuGetFramework.Parse("net46"));
+
+            var collection = new ContentItemCollection();
+            collection.Load(new string[]
+            {
+                "contentFiles/any/net4.5/config1.xml",
+                "contentFiles/any/net450/config2.xml",
+                "contentFiles/any/net4500/config3.xml",
+                "contentFiles/any/uap10.0/config4.xml",
+                "contentFiles/any/uap10.00/config5.xml",
+                "contentFiles/any/uap10.0.0/config6.xml"
+            });
+
+            // Act
+            var contentFileGroups = collection.FindItemGroups(conventions.Patterns.ContentFiles);
+
+            // Assert
+            Assert.Equal(2, contentFileGroups.Count());
+
+            Assert.Equal("net45|uap10.0", string.Join("|",
+                contentFileGroups.Select(group =>
+                    group.Properties[ManagedCodeConventions.PropertyNames.TargetFrameworkMoniker] as NuGetFramework)
+                    .Select(f => f.GetShortFolderName())
+                    .Distinct()
+                    .OrderBy(s => s)));
+        }
+
+        [Fact]
+        public void ContentFiles_InvalidPaths()
+        {
+            // Arrange
+            var conventions = new ManagedCodeConventions(
+                new RuntimeGraph(
+                    new List<CompatibilityProfile>() { new CompatibilityProfile("net46.app") }));
+
+            var criteria = conventions.Criteria.ForFramework(NuGetFramework.Parse("net46"));
+
+            var collection = new ContentItemCollection();
+            collection.Load(new string[]
+            {
+                "contentFiles/any/config1.xml",
+                "contentFiles/in.valid/net45/config2.xml",
+                "contentFiles/any/config3.xml",
+                "contentFiles/config4.xml",
+                "contentFiles",
```

Comment:
it's invalid because it doesn't have a TFM and language directory. ContentModel applies pattern matching to all of these strings, only valid matches are returned.